src/jit/lsraarm.cpp

```diff
@@ -741,7 +746,7 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
     } // end switch (tree->OperGet())
 
     // We need to be sure that we've set info->srcCount and info->dstCount appropriately
-    assert((info->dstCount < 2) || tree->IsMultiRegCall());
+    assert((info->dstCount < 2) || tree->IsMultiRegCall() || (tree->OperGet() == GT_MUL_LONG));
```

Comment:
I suspect we may add other nodes in future that return multiple registers. In any case, for clarity I think it would be useful to have a method on `GenTree()` that indicates whether it produces multiple registers, e.g. `IsMultiRegNode()`.